src/components/ServiceAlerts.js

```diff
@@ -16,32 +22,48 @@ export const ServiceAlerts = () => {
     async () => setAlerts(_.uniqWith(_.isEqual, await Ajax().Buckets.getServiceAlerts()))
   ), { ms: 60000, leading: true })
 
+  const displayNewAlerts = async () => {
+    const now = new Date()
+    const newAlerts = _.differenceWith(_.isEqual, alerts, prevAlerts)
+    const newHashes = await alertHashes(newAlerts)
+    const newAlertsWithHashes = _.filter(([, hash]) => {
+      const dismissal = getDynamic(localStorage, `dismiss-alerts/${hash}`)
+      return !dismissal || differenceInDays(parseJSON(dismissal.date), now) >= 1
+    }, _.zip(newAlerts, newHashes))
+    _.forEach(([{ link: readMoreLink, message, title, 'link-title': linkTitle, severity }, hash]) => {
+      notify(
+        severity === 'info' ? 'info' : 'warn',
+        h(Fragment, [
+          div({ style: { fontSize: 14 } }, title),
+          div({ style: { fontSize: 12, fontWeight: 500 } }, [
+            message,
+            div({ style: { marginTop: '1rem', display: 'flex' } }, [
+              readMoreLink && h(Link, { ...Utils.newTabLinkProps, href: readMoreLink, style: { fontWeight: 700 } }, [
+                linkTitle || 'Read more'
+              ]),
+              h(Link, {
+                onClick: () => {
+                  clearNotification(`alert-${hash}`)
+                  setDynamic(localStorage, `dismiss-alerts/${hash}`, { date: now })
+                },
+                style: { marginLeft: 'auto' }
+              }, ['Dismiss'])
+            ])
+          ])
+        ]),
+        { id: `alert-${hash}` }
+      )
+    }, newAlertsWithHashes)
+  }
+  const clearOutdatedAlerts = async () => {
+    const outdatedAlerts = _.differenceWith(_.isEqual, prevAlerts, alerts)
+    const outdatedHashes = await alertHashes(outdatedAlerts)
+    _.forEach(hash => clearNotification(`alert-${hash}`), outdatedHashes)
```

Comment:
I know when we were initially talking about this design, we had thought about clearing the cookies as well, for outdated alerts. This seems like maybe the place to do it? Unless there's another place that makes more sense.

Reply:
This approach relies on the automatic cleanup provided by the 'dynamic' storage mechanism. If LocalStorage fills up, these entries will be deleted starting with the oldest. I guess we could proactively remove some here, but it wouldn't catch everything, so it might be simpler to just let everything be handled the same way.